os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/example/da_ble_scatternet/ble_tizenrt_scatternet_app.c

```diff
@@ -584,16 +658,38 @@ void ble_tizenrt_scatternet_app_handle_conn_state_evt(uint8_t conn_id, T_GAP_CON
                                  disc_cause);
             }
             printf("\r\n[BLE_TIZENRT] Disconnect conn_id %d", conn_id);
-            if(GAP_CAUSE_SUCCESS == le_bond_delete_by_bd(ble_tizenrt_scatternet_app_link_table[conn_id].remote_bd,
-                            ble_tizenrt_scatternet_app_link_table[conn_id].remote_bd_type))
+            if(ble_tizenrt_scatternet_app_link_table[conn_id].role == 1)
             {
-                debug_print("\r\n[%s] Delete bond success", __FUNCTION__);
+                if(GAP_CAUSE_SUCCESS == le_bond_delete_by_bd(ble_tizenrt_scatternet_app_link_table[conn_id].remote_bd,
+                                ble_tizenrt_scatternet_app_link_table[conn_id].remote_bd_type))
+                {
+                    debug_print("\r\n[%s] Delete bond success", __FUNCTION__);
+                } else {
+                    debug_print("\r\n[%s] Delete bond fail!!!", __FUNCTION__);
```

Comment:
I don't know much about `le_bond_delete_by_bd()`, but I'm worried the memory leak or something.
Is it okay if bond is not deleted? Then when it will be deleted?

Reply:
I think it's ok. And no matter the bond is deleted or not, the connection is already disconnected, so it won't be deleted again.